### PR TITLE
REGR: CategoricalIndex(None)

### LINKED ISSUE #38944: REGR: empty CatgoricalIndex constructor by passing no data (but passing categories) no longer working
See https://github.com/pandas-dev/pandas/pull/38614/files#r551313565

https://github.com/pandas-dev/pandas/pull/38614/ changed the CategoricalIndex constructor to disallow scalar values, but at the same time it also disallowed passing *no* values to create an empty index (which in practice means passing the default of `None`):

```
In [17]: pd.CategoricalIndex(categories=['a', 'b'])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-17-6f0ae73d9374> in <module>
----> 1 pd.CategoricalIndex(categories=['a', 'b'])

~/scipy/pandas/pandas/core/indexes/category.py in __new__(cls, data, categories, ordered, dtype, copy, name)
    187         dtype: Optional[Dtype] = None,
    188         copy=False,
--> 189         name=None,
    190     ):
    191 

TypeError: CategoricalIndex(...) must be called with a collection of some kind, None was passed
```

If we make this change, this is at least something that should be deprecated IMO (eg it breaks dask parquet reading). 

But I am not sure if we should change it, since other constructors like `Series()` allow similar creation of empty objects:

cc @jbrockmendel

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #38944
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/categorical/test_constructors.py [hunk 1]
```diff
@@ -12,10 +12,14 @@
 
 class TestCategoricalIndexConstructors:
     def test_construction_disallows_scalar(self):
-        msg = "must be called with a collection of some kind"
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/categorical/test_constructors.py:15]: we still want to test this message with a scalar passed

File: pandas/tests/indexes/categorical/test_constructors.py [hunk 2]
```diff
@@ -12,10 +12,14 @@
 
 class TestCategoricalIndexConstructors:
     def test_construction_disallows_scalar(self):
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/categorical/test_constructors.py:14]: need to change the name of this test

File: pandas/tests/indexes/categorical/test_constructors.py [hunk 3]
```diff
@@ -12,10 +12,14 @@
...
-        msg = "must be called with a collection of some kind"
-        with pytest.raises(TypeError, match=msg):
+        with tm.assert_produces_warning(FutureWarning, match="without passing data"):
             CategoricalIndex(categories=list("abcd"), ordered=False)
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/categorical/test_constructors.py:16]: in #38614 some tests were changed. so I guess we should also be checking the result as well as that a warning is raised.

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jbrockmendel needs a release note mentioning the deprecated behavior

**Comment by jbrockmendel** [comment]
whatsnew note added, tests edited per suggestion

**Comment by simonjayhawkins** [comment]
Thanks @jbrockmendel

**Pull request merged**